src/cls.ts

```diff
@@ -14,12 +14,17 @@
  * limitations under the License.
  */
 
-import * as CLS from 'continuation-local-storage';
+import {Logger} from '@google-cloud/common';
+import {EventEmitter} from 'events';
 import * as semver from 'semver';
 
+import {AsyncHooksCLS} from './cls/async-hooks';
+import {AsyncListenerCLS} from './cls/async-listener';
```

Comment:
Doesn't this load `continuation-local-storage` right a way which will go and monkey patch using `async-listener`?

Reply:
That's a valid concern, but `continuation-local-storage` is only required in the constructor of `AsyncListenerCLS` (likewise for `async_hooks`).